File: objectiveai-rs/src/agent/completions/response/unary/message.rs
//! Message type for unary agent completion responses.

use crate::agent::completions::response;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(untagged)]
pub enum Message {
    Assistant(super::AssistantResponse),
    Tool(response::ToolResponse),
}

impl From<response::streaming::MessageChunk> for Message {
    fn from(chunk: response::streaming::MessageChunk) -> Self {
        match chunk {
            response::streaming::MessageChunk::Assistant(chunk) => {
                Message::Assistant(chunk.into())
            }
            response::streaming::MessageChunk::Tool(chunk) => {
                Message::Tool(chunk)
            }
        }
    }
}

File: objectiveai-js/src/agent/completions/response/unary/message.ts
import { z } from "zod";
import { AgentCompletionsResponseToolResponseSchema } from "../toolResponse";
import { AgentCompletionsResponseUnaryAssistantResponseSchema } from "./assistantResponse";

export const AgentCompletionsResponseUnaryMessageSchema = z.union([AgentCompletionsResponseUnaryAssistantResponseSchema, AgentCompletionsResponseToolResponseSchema]).meta({ title: "agent.completions.response.unary.Message" });
export type AgentCompletionsResponseUnaryMessage = z.infer<typeof AgentCompletionsResponseUnaryMessageSchema>;
